src/replica/GetStatusQservMgtRequest.cc

```diff
@@ -50,86 +40,51 @@ LOG_LOGGER _log = LOG_GET("lsst.qserv.replica.GetStatusQservMgtRequest");
 
 namespace lsst::qserv::replica {
 
-GetStatusQservMgtRequest::Ptr GetStatusQservMgtRequest::create(
-        ServiceProvider::Ptr const& serviceProvider, string const& worker,
+string taskSelectorToHttpQuery(wbase::TaskSelector const& taskSelector) {
+    string query;
+    query += "?include_tasks=" + string(taskSelector.includeTasks ? "1" : "0");
+    query += "&max_tasks=" + to_string(taskSelector.maxTasks);
+    if (!taskSelector.queryIds.empty()) {
+        ostringstream ss;
+        copy(taskSelector.queryIds.begin(), taskSelector.queryIds.end() - 1,
+             ostream_iterator<qserv::QueryId>(ss, ","));
+        ss << taskSelector.queryIds.back();
+        query += "&query_ids=" + ss.str();
+    }
+    if (!taskSelector.taskStates.empty()) {
+        ostringstream ss;
+        copy(taskSelector.taskStates.begin(), taskSelector.taskStates.end() - 1,
+             ostream_iterator<wbase::TaskState>(ss, ","));
+        ss << taskSelector.taskStates.back();
+        query += "&task_states=" + ss.str();
+    }
+    return query;
+}
+
+shared_ptr<GetStatusQservMgtRequest> GetStatusQservMgtRequest::create(
+        shared_ptr<ServiceProvider> const& serviceProvider, string const& worker,
         wbase::TaskSelector const& taskSelector, GetStatusQservMgtRequest::CallbackType const& onFinish) {
-    return GetStatusQservMgtRequest::Ptr(
+    return shared_ptr<GetStatusQservMgtRequest>(
```

Comment:
I started disliking `::Ptr` since it requires extra includes to be done in the header files for dependent classes that do not need to be elaborated in the headers. here is one example. Suppose you have a header where you have:
```
// File: A.h

// Many other headers are included below. The headers add numerous compilation dependencies
// which need to be processed by the compiler.
#include ....

// Now the class
class A {
public:
    typedef std::shared_ptr<A> Ptr;
    // Lots of heavy stuff goes below
   ...
};
```
And then you have the dependent header where we include the first one only to get that A::Ptr
```
// File: B.h
#include "A.h"
...
```
A much better way (if the forward declaration would work) would be to:
```
// File: B.h
class A:

// And then we can use std::shared_ptr<A> instead of A::Ptr
```
At this point my logic was - why do I need that `A::Ptr` at all everywhere if I can live with `shared_ptr<A>`. So, I started (gradually) getting rid of those `::Ptr` since they don't improve the code and just introduce an alternative way of referencing instances. Besides, we have `unique_ptr<A>` and `A*` which are also pointers. It's probably better to be specific on the kind of pointer used in each case.

A drawback of this "philosophy" is some increase in the amount of code.

Reply:
Forgot to mention the second reason - an attempt to speed up the compilation by making headers lighter (not pulling unneeded dependencies simply because it might be convenient to use `A::Ptr` instead of `shared_ptr<A>`).